api/src/main/java/org/pmiops/workbench/initialcredits/InitialCreditsExpirationServiceImpl.java

```diff
@@ -17,4 +67,64 @@ public Optional<Timestamp> getCreditsExpiration(DbUser user) {
         // .filter(() -> institutionService.shouldBypassForCreditsExpiration(user))
         .map(DbUserInitialCreditsExpiration::getExpirationTime);
   }
+
+  private void checkExpiration(DbUser user, Timestamp now) {
+    DbUserInitialCreditsExpiration userInitialCreditsExpiration =
+        user.getUserInitialCreditsExpiration();
+    if (null != userInitialCreditsExpiration
+        && !userInitialCreditsExpiration.isBypassed()
+        && userInitialCreditsExpiration
+        .getNotificationStatus()
+        .equals(NotificationStatus.NO_NOTIFICATION_SENT)
+        && !(userInitialCreditsExpiration.getExpirationTime().after(now))) {
```

Comment:
suggest to add a info logging here about a user's credit expired. It would be helpful for future support and debugging.